src/components/modals/ButtonsModal.tsx

```diff
@@ -54,9 +54,9 @@ export interface ButtonModalProps<Buttons> {
  * Build a custom modal component if you need form fields, check boxes,
  * or other interactive elements.
  */
-export function ButtonsModal<Buttons extends Record<string, ButtonInfo>>(
-  props: ButtonModalProps<Buttons>
-) {
+export const ButtonsModal: React.FC<
+  ButtonModalProps<Record<string, ButtonInfo>>
+> = (props: ButtonModalProps<Record<string, ButtonInfo>>) => {
```

Comment:
You don't need `(props: ...) => {` because the types are already on the left. Just do `props =>`.